openmp/device/src/Allocator.cpp

```diff
@@ -23,7 +23,7 @@ using namespace allocator;
 // Provide a default implementation of malloc / free for AMDGPU platforms built
 // without 'libc' support.
 extern "C" {
-#if defined(__AMDGPU__) && !defined(OMPTARGET_HAS_LIBC)
+#if (defined(__AMDGPU__) || defined(__SPIRV__)) && !defined(OMPTARGET_HAS_LIBC)
```

Comment:
i think it's better to have a define in the dependent hedears that like NEEDS_ALLOC than have a list of architectures in independent files. Also, update the comment above.